File: FrontEnd/MoodTracker/Model/Entry.swift
//
//  Model.swift
//  MoodTracker
//
//  Created by Chrissy Vinco on 7/30/22.
//

import Foundation

struct Entry: Codable {
    var id: String?
    var dateCreated: Double? 
    var time: Date = Date()
    var mood: Float? = 5.0
    var note: String?
    var tags: [Tag]? = []
}

struct Username: Codable {
    var username: String
}

File: Backend/Models/Entry.js
import mongoose from 'mongoose';
// import { Tag } from './Tag';

const EntrySchema = new mongoose.Schema({ 
    dateCreated: {
        type: Number,
        immutable: true
    }, 
    time: {
        type: Date, 
        default: () => Date.now(),
        immutable: true
    },
    mood: {
        type: Number,
    }, 
    note: {
        type: String,
        maxLength: 250,
        minLength: 0
    }, 
    user: {
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'User'
    },
    tags: [{
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Tag'
    }]
})

const Entry = mongoose.model("Entry", EntrySchema);
export { Entry }